lisp/ess-custom.el

```diff
@@ -1781,6 +1781,9 @@ for editing and then to be returned to the process.")
 The other variables ...-program should be changed, for the
 corresponding program.")
 
+(make-obsolete-variable 'inferior-ess-start-args
+                        "Use the language specific variables"
```

Comment:
The message can probably be a bit more helpful like, "Use language-specific variables instead like inferior-R-args".

Reply:
I considered this as well, but I don't like the non-standard namespace here. Should we alias to `inferior-ess-r-start-args`?

Reply:
Might as well rename them all while we're at it so that they're more standard. I'd do something like `ess-<lang>-start-args`.

Reply:
Should we have the convention that "inferior" is a technical term for ESS developers that shouldn't leak into the UI?

Dev oriented: `inferior-ess-r-force`
User oriented: `ess-r-start-args`

As a particle, `inferior` suggests we're interacting with the associated process of the current buffer.

Reply:
Makes sense to me :+1:

Reply:
I'm now wondering if we should take the renaming opportunity to update the interface so that `start-args` variables should be `nil` or a list of strings. This might simplify things and make it easier to combine arguments from different sources.

For instance we'd set the default to `'("--no-readline")` and users would add new arguments in their config with `add-to-list`. Then we no longer need `ess-R-readline` in our config UI.

Reply:
I never liked inferior either, but I think it's useful to distinguish those variables if they are strictly relevant to the process. `repl-` would have been better.

Reply:
Completely agree on the argument list BTW.

Reply:
Right, `repl-` or `iess-` would be shorter. But I think for the UI `ess-` should mean "interaction with REPL". For other functionality (help, package, etc) we have other namespaces.

Reply:
`inferior-` has the advantage of being shared across other Emacs packages (e.g. inferior-python-mode)

Reply:
Do we agree that `inferior` should not be exposed to end users in UI variables and functions? For instance, I was planning to rename `inferior-ess-reload` to just `ess-reload`. It is shorter to type, less scary, and more easily searchable on `M-x`.

Reply:
I don't think so. That would require renaming inferior-ess-r-mode and the others since they generate e.g. inferior-ess-r-mode-hook, -map, etc. I'm open to arguments to change my mind, though.

Reply:
I am fine with user commands not containing `inferior`, but something should be there in this particular case. `ess-reload` is too generic. `ess-reload-process` or `ess-reload-repl` maybe?

There are 2 more commands which we might consider renaming then: `inferior-ess-force` and `inferior-ess-send-input`.

Reply:
@jabranham, I think Lionel didn't  mean the mode functions. Renaming those would be troublesome indeed.

Reply:
Good point about the hooks, though I think it would be ok to just alias the hooks without renaming everything.

> There are 2 more commands which we might consider renaming then: inferior-ess-force and inferior-ess-send-input.

I would think these commands are more on the programming side, so I would not rename them.

Let's keep thining about this then.

Reply:
FWIW `inferior-ess-force` is already aliased to `ess-force-buffer-current`

Reply:
(which is also not a great name, if you ask me) :-)

Reply:
We could stick with `-process` in our names for such functionality: `ess-reload-process`, `ess-force-current-process`.

Reply:
hmm maybe. In the particular case of reloading, I think we should follow the existing naming scheme so it's consistent with `ess-quit`.

Reply:
I would generally prefer to be more explicit, but ok. If we leave it without `-process` then how about `-restart` rather than `-reload`? It's a bit more specific and sesman (which I will bring to ESS at some point) already has `sesman-restart` command.

Reply:
`ess-restart` sounds good to me.